applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.cpp

```diff
@@ -0,0 +1,390 @@
+        const double mu_max = mEnergyConstant * std::sqrt(h2) * inf_norm.Density * inf_norm.TotalVelocity;
+        const double mu_h = std::min(mu_e, mu_max);
+
+        const double mu_rho = mArtificialMassDiffusivityPrandtl * mu_h / inf_norm.Density;
+        const double kappa  = mArtificialConductivityPrandtl * mu_h / (heat_capacity_ratio - 1.0);
+
+        DistributeVariablesToNodes(r_element, mu_h, mu_rho, kappa);
+    });
+
+    KRATOS_CATCH("")
+}
+
+
+void ShockCapturingEntropyViscosityProcess::DistributeVariablesToNodes(
+    Element& rElement,
+    const double ArtificialBulkViscosity,
+    const double ArtificialMassDiffusivity,
+    const double ArtificialConductivity) const
+{
+    auto& r_geometry = rElement.GetGeometry();
+    const double element_volume = r_geometry.LocalSpaceDimension() == 3 ? r_geometry.Volume() : r_geometry.Area();
+
+    for(unsigned int i=0; i<r_geometry.size(); ++i)
+    {
+        auto& r_node = r_geometry[i];
+        const double weight = element_volume / r_node.GetValue(NODAL_AREA);
+        r_node.SetLock();
+        r_node.GetValue(ARTIFICIAL_BULK_VISCOSITY) += weight * ArtificialBulkViscosity;
+        r_node.GetValue(ARTIFICIAL_MASS_DIFFUSIVITY) += weight * ArtificialMassDiffusivity;
+        r_node.GetValue(ARTIFICIAL_CONDUCTIVITY) += weight * ArtificialConductivity;
+        r_node.UnSetLock();
+    }
+}
+
+
+double ShockCapturingEntropyViscosityProcess::ComputeEntropy(
+    const double Density,
+    const double Pressure,
+    const double Gamma)
+{
+    return Density / (Gamma - 1.0) * std::log(Pressure / std::pow(Density, Gamma));
+}
+
+
+double ShockCapturingEntropyViscosityProcess::ComputeHSquared(const Element& rElement)
+{
+    double h_squared = std::numeric_limits<double>::max();
+    const auto& r_geometry = rElement.GetGeometry();
+
+    // H is the shortest edge of the element
+    for(unsigned int i=0; i<r_geometry.size(); ++i)
+    {
+        const unsigned j = (i + 1) % r_geometry.size();
+        const auto edge = r_geometry[j] - r_geometry[i];
+        h_squared = std::min(h_squared, inner_prod(edge, edge));
+    }
+
+    return h_squared;
+}
+
+
+ShockCapturingEntropyViscosityProcess::InfNormData ShockCapturingEntropyViscosityProcess::ComputeElementalInfNormData(
+    const Element& rElement,
+    const double DeltaTime,
+    const double HeatCapacityRatio)
+{
+    TotalDerivativeUtil entropy_td;
+    TotalDerivativeUtil density_td;
+    Vector total_velocities;
+
+    std::tie(entropy_td, density_td, total_velocities) = BuildTotalDerivativeUtils(rElement, DeltaTime, HeatCapacityRatio);
+    return ComputeInfNorms(rElement.GetGeometry(), entropy_td, density_td, total_velocities);
+}
+
+
+std::tuple<ShockCapturingEntropyViscosityProcess::TotalDerivativeUtil, ShockCapturingEntropyViscosityProcess::TotalDerivativeUtil, Vector>
+ShockCapturingEntropyViscosityProcess::BuildTotalDerivativeUtils(const Element& rElement, const double DeltaTime, const double HeatCapacityRatio)
```

Comment:
```suggestion
ShockCapturingEntropyViscosityProcess::BuildTotalDerivativeUtils(
    const Element& rElement,
    const double DeltaTime,
    const double HeatCapacityRatio)
```
(minor style guide suggestion)